apply/apply.go

```diff
@@ -68,7 +69,14 @@ func execPipeline(ctx context.Context, in *graph.Graph, pipelineF MkPipelineF, r
 		notify.Transform(func(id string, out *graph.Graph) error {
 			renderingPlant.Graph = out
 			pipeline := pipelineF(out, id)
-			val, pipelineError := pipeline.Exec(out.Get(id))
```

Comment:
inside of `apply/pipeline` we have:

```
type pipelineGen struct {
	Graph          *graph.Graph
	ID             string
	RenderingPlant *render.Factory
}
```
and 
```
func Pipeline(g *graph.Graph, id string, factory *render.Factory) executor.Pipeline {
	gen := &pipelineGen{Graph: g, RenderingPlant: factory, ID: id}
````
since the ID is now going to be part of the element we could remove the ID from `pipelineGen` and the call to `Pipeline`.  It's not necessarily hurting anything to have it as it is, but it does get us down to a single source-of-truth for the ID.